apps/studio/src/components/export/ExportNotification.vue

```diff
@@ -28,47 +31,53 @@ export default {
   },
   computed: {
     notificationText() {
-      const exportName = this.exporter.table ? this.exporter.table.name : this.exporter.queryName;
       // CoPilot suggested the comment below, and it was right af lol
       // this is a hack to get the countExported to update
       const countExported = this.countExported;
       const percentComplete = this.percentComplete;
-      return this.exporter.table
-        ? `(${percentComplete}%) Exporting table '${exportName}'`
-        : `(${countExported} rows) Exporting query '${exportName}'`
+      return percentComplete
+        ? `(${percentComplete}%) Exporting table '${this.exportName}'`
+        : `(${countExported} rows) Exporting query '${this.exportName}'`
     },
   },
   methods: {
-    cancelExport() {
-      if (!this.exporter) {
+    async cancelExport() {
+      if (!this.exportId) {
         return;
       }
-      this.exporter.abort();
+      await this.$server.send('export/cancel', { id: this.exportId });
       this.notification.close();
-      const exportName = this.exporter.table ? this.exporter.table.name : this.exporter.queryName;
+      const exportName = await this.$server.send('export/name', { id: this.exportId });
       this.$noty.error(`${exportName} export aborted`);
     },
-    updateProgress(progress) {
+    async updateExportName() {
+      this.exportName = await this.$server.send('export/name', { id: this.exportId });
+    },
+    async updateProgress(progress) {
       // not quite sure why this hackiness (and only this hackiness) finally made it work but i'll take it
       this.countExported = progress.countExported
-      this.percentComplete = this.exporter.table ? progress.percentComplete : progress.countExported
+      this.percentComplete = progress.percentComplete ? progress.percentComplete : progress.countExported
     }
   },
   watch: {
     notificationText: {
       handler() {
         if (this.notification) {
-          this.notification.setText(this.notificationText);
+          this.notification?.setText(this.notificationText);
         }
       },
     },
   },
-  mounted() {
-    this.exporter.onProgress(this.updateProgress)
-    this.notification.show();
+  async mounted() {
```

Comment:
@rathboma Okay so, this part actually doesn't handle errors, it handles progress, was annoying to get working lol.

For errors, just like any handler, if the underlying function throws, it will be sent back to the client as an 'error' (see utility.ts `runHandler`). Once the client receives a reply, if it's an error it will reject the promise it returned with the error, which is then handled through whatever channels we want